core/src/main/java/org/apache/calcite/sql/type/SqlTypeTransforms.java

```diff
@@ -98,6 +98,17 @@ public abstract class SqlTypeTransforms {
           opBinding.getTypeFactory().createTypeWithNullability(
               requireNonNull(typeToTransform, "typeToTransform"), true);
 
+  /**
+   * Parameter type-inference transform strategy where a derived type is
+   * transformed into the same type with nulls allowed if the result is not floating point.
+   */
+  public static final SqlTypeTransform FORCE_NULLABLE_NON_FP =
+      (opBinding, typeToTransform) -> {
+        boolean nullable = !SqlTypeName.APPROX_TYPES.contains(typeToTransform.getSqlTypeName());
```

Comment:
For approximate types `nullable` becomes `false` which will end up making the resulting type non-nullable which I don't think its correct. I have the impression that the result of the division is always nullable and this is also unrelated to the fact that we divide by zero or not.
**Example**
```sql
2.5/null
null/2.5
(float) null/ (float) null
```